asmcomp/amd64/arch.ml

```diff
@@ -13,13 +13,25 @@
 (*                                                                        *)
 (**************************************************************************)
 
+(* LZCNT instruction is not avialble on Intel Architecures prior to Haswell.
+
+   Important: lzcnt assembles to bsr on architectures prior to Haswell.  Code
+   that uses lzcnt will run on older Intels and silently produce wrong
+   results.
+
+   This is a placehold. Replace with architecture-specific logic in
+   command-line options and cpuid checking. *)
+let lzcnt_support = ref false
+
 (* Machine-specific command-line options *)
 
 let command_line_options =
   [ "-fPIC", Arg.Set Clflags.pic_code,
       " Generate position-independent machine code (default)";
     "-fno-PIC", Arg.Clear Clflags.pic_code,
-      " Generate position-dependent machine code" ]
+      " Generate position-dependent machine code";
+    "-flzcnt", Arg.Set lzcnt_support,
+      " Use lzcnt instruction to count leading zeros" ]
```

Comment:
Formatting suggestion: add trailing `;` and put the `]` on its own line (so that the next flag addition results in a nicer diff).

Reply:
Fixed.